### ISSUE TITLE
Add more customization abilities regarding optimizers and loss functions

### ISSUE THREAD

**Issue by nclgbd** [nclgbd/PyTorch-Utilities#20]
It's pretty standard to use `CrossEntropyLoss` and `Adam` as our loss function and optimizer respectively for computer vision tasks, however with the research everchanging it is worth adding in the ability to customize these features, along with adding in any other optimization methods from the `torch.optim` library.
